### PR TITLE
Recipe app

### PR DESCRIPTION
**Pull request by AHP15**
In this PR, I added with my coding partner @kennankole the following:

- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
- Should be built with Devise.
- Food list:
- Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
- Should lead to a form that allows users to add new food.
- Recipes list:
- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
- Should display a list of all public recipes ordered by newest as in the wireframe.
- Should lead to recipe details.
- Recipe details:
- Should display a toggle button that allows for a recipe to be made public or private.
- If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view (only for a group that has less than 3 members):
- Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- Should count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.
- Write unit and integration tests

Cheers 👍

### CODE REVIEW

File: app/models/food.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Food < ApplicationRecord
+  belongs_to :user, class_name: 'User'
+
```
Comment (Kamba56) [inline app/models/food.rb:3]: - [ ] The relationship between recipes and foods is a many-to-many relationship described by the recipe_foods class. Kindly add this relationship to your model.
in food class add `has_many :recipe_foods, class_name: :recipe_foods`
`has_many :foods, through: :recipe_foods`
Kindly check [here](https://guides.rubyonrails.org/association_basics.html#the-has-many-through-association) for more details.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @Abdessittir, @kennankole , 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great use of Git workflow!👏👏👏
- [x] All linters passed
- [x] Great job using devise for authentication
- [x] Great job implementing the wireframes

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by arslanbisharat** [comment]
Hi @Abdessittir,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by arslanbisharat** [comment]
I mistakenly marked the project "Requested Changed" instead of approved. I would like to apologize for my mistake. Kindly request a new review and CR will take care of it for you. 

Muhammad Arslan

**Comment by Abdessittir** [comment]
Hi future reviewer, Tthe previous reviewer mistakenly marked the project "Requested Changed" instead of approved kindly approve my pull request

**Review by BertrandConxy — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- good PR description
- correct flow is used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**